### PR TITLE
Feature/lwc spec cfd

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP

- [x] SignatureProcessModal 
- [x] WithdrawalModal
- [x] DepositModal
- [x] PositionOpenModal 
- [x] PositionUpdatedModal
- [x] PositionCloseModal  
- [x]  CandlestickChart
- [x] globalContext
- [x] userContext
- [x] BalanceSection
- [x] InterestSection
- [x] PnlSection

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 0
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/components/position_open_modal/position_open_modal.tsx [hunk 1]
```diff
@@ -93,48 +111,71 @@ const PositionOpenModal = ({
...
+        globalCtx.dataSuccessfulModalHandler({
+          modalTitle: 'Open Position',
+          modalContent: 'Transaction succeed',
+          btnMsg: 'View on Boltchain',
+          btnUrl: '#',
+        });
+
+        globalCtx.visibleSuccessfulModalHandler();
+      } else if (
+        // ToDo: Expired quotation [Failed] and Rejected signature [Canceled]
```
Comment (arealclimber) [inline src/components/position_open_modal/position_open_modal.tsx:143]: Expired quotation [Canceled] Error code 1 -> 過期後 context 提醒 component 這張報價單過期
Expired quotation [Failed] Error code 2 -> 過期後雖然顯示 canceled，但用戶依然在 Metamask 簽名

File: src/components/position_open_modal/position_open_modal.tsx [hunk 2]
```diff
@@ -93,48 +111,71 @@ const PositionOpenModal = ({
...
+        result.code === Code.INTERNAL_SERVER_ERROR ||
+        result.code === Code.INVAILD_INPUTS ||
+        result.code === Code.SERVICE_TERM_DISABLE ||
+        result.code === Code.WALLET_IS_NOT_CONNECT
+      ) {
```
Comment (arealclimber) [inline src/components/position_open_modal/position_open_modal.tsx:148]:
```result.code === Code.SERVICE_TERM_DISABLE ||
        result.code === Code.WALLET_IS_NOT_CONNECT```

要顯示 canceled

File: src/components/position_open_modal/position_open_modal.tsx [hunk 3]
```diff
@@ -93,48 +111,71 @@ const PositionOpenModal = ({
...
+          modalContent: 'Transaction succeed',
+          btnMsg: 'View on Boltchain',
+          btnUrl: '#',
+        });
+
+        globalCtx.visibleSuccessfulModalHandler();
+      } else if (
+        // ToDo: Expired quotation [Failed] and Rejected signature [Canceled]
+        result.code === Code.INTERNAL_SERVER_ERROR ||
+        result.code === Code.INVAILD_INPUTS ||
```
Comment (arealclimber) [inline src/components/position_open_modal/position_open_modal.tsx:145]:
```
result.code === Code.INTERNAL_SERVER_ERROR ||
result.code === Code.INVAILD_INPUTS ||
```

都是資料格式不合格

File: src/components/withdrawal_modal/withdrawal_modal.tsx [hunk 4]
```diff
@@ -83,58 +84,140 @@ const WithdrawalModal = ({
...
-      globalCtx.visibleFailedModalHandler();
+      // ToDo: Rejected signature [Canceled]
+      // else if
+    } catch (error: any) {
+      globalCtx.eliminateAllModals();
+
+      // Info: Signature rejected
+      // ToDo: Catch the error which user rejected the signature in UserContext (20230411 - Shirley)
+      if (error?.code === 4001) {
+        globalCtx.dataCanceledModalHandler({
```
Comment (arealclimber) [inline src/components/withdrawal_modal/withdrawal_modal.tsx:152]: - 寫在 user context

File: src/components/withdrawal_modal/withdrawal_modal.tsx [hunk 5]
```diff
@@ -83,58 +84,140 @@ const WithdrawalModal = ({
...
+          modalTitle: 'Withdraw',
+          modalContent: 'Transaction canceled',
+        });
+
+        globalCtx.visibleCanceledModalHandler();
+      } else {
+        // Info: Unknown error
+        globalCtx.dataFailedModalHandler({
```
[the remark this answers is not in the document]
Comment (arealclimber) [reply src/components/withdrawal_modal/withdrawal_modal.tsx:168]: throw CustomError(withdraw_failed)

File: src/components/signature_process_modal/signature_process_modal.tsx [hunk 6]
```diff
@@ -157,45 +121,53 @@ const SignatureProcessModal = ({
...
-      // TODO1
-      unlock();
+        const signResult = await userCtx.signServiceTerm();
 
-      if (signResult) {
-        setConnectingProcess(ConnectingProcess.CONNECTED);
+        unlock();
+
+        if (signResult.success) {
+          setConnectingProcess(ConnectingProcess.CONNECTED);
```
Comment (arealclimber) [inline src/components/signature_process_modal/signature_process_modal.tsx:144]: 第二部簽名失敗的 reason
```
Something went wrong, please try again
```

File: src/components/position_open_modal/position_open_modal.tsx [hunk 7]
```diff
@@ -96,48 +111,103 @@ const PositionOpenModal = ({
...
+
+        globalCtx.visibleCanceledModalHandler();
+      } else {
+        // ToDo: Throw error (20230412 - Shirley)
+        // result.code === Code.CFD_OPEN_FAILED
+        // Info: Unknown error between context and component
+        globalCtx.dataFailedModalHandler({
+          modalTitle: t('POSITION_MODAL.OPEN_POSITION_TITLE'),
+          failedTitle: t('POSITION_MODAL.FAILED_TITLE'),
+          failedMsg: t('POSITION_MODAL.FAILED_REASON_FAILED_TO_OPEN'),
```
Comment (arealclimber) [inline src/components/position_open_modal/position_open_modal.tsx:206]: 顯示對應的 error 訊息

Comment (arealclimber) [reply src/components/position_open_modal/position_open_modal.tsx:206]: 之後要在這邊收集 error 訊息回傳到系統那邊

File: src/components/withdrawal_modal/withdrawal_modal.tsx [hunk 8]
```diff
@@ -83,58 +84,152 @@ const WithdrawalModal = ({
...
+          modalTitle: t('D_W_MODAL.WITHDRAW'),
+          failedTitle: t('D_W_MODAL.FAILED_TITLE'),
+          failedMsg: t('D_W_MODAL.FAILED_REASON_FAILED_TO_WITHDRAW'),
+        });
+
+        globalCtx.visibleFailedModalHandler();
+      }
     }
+    // const result = await userCtx.withdraw({
+    //   orderType: OrderType.WITHDRAW,
```
Comment (arealclimber) [inline src/components/withdrawal_modal/withdrawal_modal.tsx:182]: 刪掉 comments

File: src/components/withdrawal_modal/withdrawal_modal.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (arealclimber) [reply src/components/withdrawal_modal/withdrawal_modal.tsx:168]: 這邊不需要 throw Error，需要做 1. 通知用戶失敗 2. 之後回報錯誤訊息給後端

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**